### PR TITLE
add harbor cluster reconcile logic freamwork.

### PR DESCRIPTION
**Pull request by wangcanfengxs**

### CODE REVIEW

File: controllers/harborcluster_controller.go [hunk 1]
```diff
@@ -37,14 +38,100 @@ type HarborClusterReconciler struct {
 // +kubebuilder:rbac:groups=goharbor.goharbor.io,resources=harborclusters/status,verbs=get;update;patch
 
 func (r *HarborClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("harborcluster", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("harborcluster", req.NamespacedName)
 
-	// your logic here
+	var harborCluster goharborv1.HarborCluster
```
Comment (steven-zou) [inline controllers/harborcluster_controller.go:44]:
```suggestion
	var harborCluster *goharborv1.HarborCluster
```
???

Comment (wangcanfengxs) [reply controllers/harborcluster_controller.go:50]: I think, here needs a HarborCluster Object, not an address ref to HarborCluster Object.  Even if using a pointer,   you may use DeepCopy !

Comment (steven-zou) [reply controllers/harborcluster_controller.go:50]: You declare an object, but when using in the code, they are all ref point (& used), why not directly declare a pointer?

File: lcm/lcm.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package lcm
+
+import v1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+
+// This package container interface of harbor cluster service lifecycle manage.
+
+type Controller interface {
+	// Provision the new dependent service by the related section of cluster spec.
+	provision(spec *v1.HarborCluster) (*CRStatus, error)
```
Comment (steven-zou) [inline lcm/lcm.go:9]: Why define the unexported methods in the interface?

Comment (steven-zou) [reply lcm/lcm.go:9]: The implementation of the interface and the interface definition itself may be located at different packages.

Comment (wangcanfengxs) [reply lcm/lcm.go:9]: I think this interface is always invoked in Reconcile. The implemenets of Reconciler interface are also implement the Controller interface, then we can use methods in Reconcile.

File: controllers/harborcluster_controller.go [hunk 3]
```diff
@@ -17,18 +17,25 @@ package controllers
...
-	goharborv1 "src/github.com/goharbor/harbor-cluster-operator/api/v1"
+	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
 )
 
 // HarborClusterReconciler reconciles a HarborCluster object
 type HarborClusterReconciler struct {
 	client.Client
+	ServiceGetter
 	Log    logr.Logger
 	Scheme *runtime.Scheme
```
Comment (Ghostbaby) [inline controllers/harborcluster_controller.go:40]: maybe we should add the event record, e.p: `Recorder record.EventRecorder`

**Pull request merged**